### PR TITLE
standard compliant JCS implementation

### PR DESCRIPTION
**Pull request by evik42**
Resolves: #10

### CODE REVIEW

File: jcs/build.rs
[the remark this answers is not in the document]
Comment (evik42) [reply jcs/build.rs:1]: The goal of that test is to test _all_ files in the testdata input directory. It can be accomplished via test_cases also but then extra care needs to be taken to make sure that all files are tested, this is also true when adding new files to cover additional edge cases.

File: jcs/src/jcs.rs [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+#[inline]
+pub fn to_vec<S: Serialize>(value: &S) -> serde_json::Result<Vec<u8>> {
+    let mut buffer = Vec::with_capacity(1024);
+    to_writer(value, &mut buffer).map(|_| buffer)
+}
+
+#[inline]
+pub fn to_writer<S: Serialize, W: io::Write>(value: &S, writer: &mut W) -> serde_json::Result<()> {
+    value.serialize(&mut JcsSerializer::new(writer))
+}
```
[the remark this answers is not in the document]
Comment (evik42) [reply jcs/src/jcs.rs:30]: I leave this exercise to the one who will create an external lib from this.

File: jcs/src/jcs.rs
[the remark this answers is not in the document]
Comment (evik42) [reply jcs/src/jcs.rs:1]: No, I basically copied the code from the existing serde_json library and modified to match our needs.
It is the same with the `unsafe` block.

**Comment by evik42** [comment]
Moving out to a separate library
